### ISSUE TITLE
aspectRatio() should return double/float not <T>

### ISSUE THREAD

**Issue by kghose** [mosra/magnum#251]
https://github.com/mosra/magnum/blob/d9b4207426022e3444543d34ece06b2c4e1daed3/src/Magnum/Math/Vector2.h#L175

The reason for this is that it causes less user surprise.

My use was that I was writing:

```
void GrohoApp::viewportEvent(const Vector2i& size)
{
    GL::defaultFramebuffer.setViewport(Range2Di{ { 0, 0 }, size });
    camera.projection = Matrix4::perspectiveProjection(
                            35.0_degf, size.aspectRatio(), 0.01f, 100.0f)
        * Matrix4::translation(Vector3::zAxis(-10.0f));

    redraw();
}
```
I expected `size.aspectRatio()` to return a `float`/`double` which in my opinion makes the most sense given its expected uses.

Thanks!

**Comment by kghose** [comment]
@mosra I can see the "types" logic of casting it to a fp vector, however, I suspect that the use of `aspectRatio()` is largely as I've used it, in which case a user is expecting a **ratio**, which is a floating point value.

**Comment by mosra** [comment]
Oh, I fixed this two weeks ago and forgot to mention it here. See 04f4917f7b31fe14c53a4f82fd893b261dd6cca3 for details -- basically I disabled all operations that don't make sense on integer types and left them only on floating-point types. This is consistent with for example bitwise operators that are disabled for floating-point types. Your proposed solution would bring some ambiguities (for example, what it should return for a vector of 64-bit integers? a float? a double? a long double? and what for 8-bit integers?) and so this felt more straighforward to implement and document. Hope that's okay :)
